CrossPlatformDesktopProject/CrossPlatformDesktopProject/GameStates/RoomTransitionState.cs

```diff
@@ -2,33 +2,226 @@
+                rightDoor2 = RoomTextureStorage.RIGHT_OPEN_DOOR;
+            }
         }
 
         public void Draw(SpriteBatch sb)
         {
-
-            Rectangle window = new Rectangle(0, 0, game.GraphicsDevice.Viewport.Width, game.GraphicsDevice.Viewport.Height);
-            sb.Draw(screenshot, window, window, Color.White);
+            sb.Draw(texture, borderDestination1, borderSource, Color.White);
+            sb.Draw(texture, borderDestination2, borderSource, Color.White);
+            sb.Draw(texture, destinationRoom1, sourceRoom1, Color.White);
+            sb.Draw(texture, destinationRoom2, sourceRoom2, Color.White);
+            sb.Draw(texture, topDoorDest1, topDoor1, Color.White);
+            sb.Draw(texture, bottomDoorDest1, bottomDoor1, Color.White);
+            sb.Draw(texture, leftDoorDest1, leftDoor1, Color.White);
+            sb.Draw(texture, rightDoorDest1, rightDoor1, Color.White);
+            sb.Draw(texture, topDoorDest2, topDoor2, Color.White);
+            sb.Draw(texture, bottomDoorDest2, bottomDoor2, Color.White);
+            sb.Draw(texture, leftDoorDest2, leftDoor2, Color.White);
+            sb.Draw(texture, rightDoorDest2, rightDoor2, Color.White);
+            
         }
 
         public void Update()
         {
+            switch (mySide)
+            {
+                case CollisionSides.Right:
+                    borderDestination1 = new Rectangle(scrolling, 0, borderWidth, borderHeight);
+                    borderDestination2 = new Rectangle(scrolling + borderWidth, 0, borderWidth, borderHeight);
+                    destinationRoom1 = new Rectangle((borderWidth - roomWidth) / 2 + scrolling, (borderHeight - roomHeight) / 2, roomWidth, roomHeight);
+                    destinationRoom2 = new Rectangle((borderWidth - roomWidth) / 2 + scrolling + borderWidth, (borderHeight - roomHeight) / 2, roomWidth, roomHeight);    
+                    topDoorDest1 = new Rectangle(borderWidth / 2 - doorSize / 2 + scrolling, 0, doorSize, doorSize);
+                    leftDoorDest1 = new Rectangle(scrolling, borderHeight / 2 - doorSize / 2, doorSize, doorSize);
+                    bottomDoorDest1 = new Rectangle(borderWidth / 2 - doorSize / 2 + scrolling, borderHeight - doorSize, doorSize, doorSize);
+                    rightDoorDest1 = new Rectangle(borderWidth - doorSize + scrolling, borderHeight / 2 - doorSize / 2, doorSize, doorSize);
+                    topDoorDest2 = new Rectangle(borderWidth / 2 - doorSize / 2 + borderWidth + scrolling, 0, doorSize, doorSize);
+                    leftDoorDest2 = new Rectangle(scrolling + borderWidth, borderHeight / 2 - doorSize / 2, doorSize, doorSize);
+                    bottomDoorDest2 = new Rectangle(borderWidth / 2 - doorSize / 2 + borderWidth + scrolling, borderHeight - doorSize, doorSize, doorSize);
+                    rightDoorDest2 = new Rectangle(borderWidth - doorSize + borderWidth + scrolling, borderHeight / 2 - doorSize / 2, doorSize, doorSize);
```

Comment:
I wonder if this can be factored out into a for-each loop.

Reply:
Yeah, it all looks fine but a refactoring consideration could be to try to cut out some repetition if that's possible idk.